cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -0,0 +1,532 @@
+                                       "container/non-existentdir"));
+  ASSERT_RAISES(IOError, fs_->CopyFile("container/somedir/subdir/subfile",
+                                       "container/somedir/subdir"));
+  ASSERT_RAISES(IOError, fs_->CopyFile("container/somedir/subdir/subfile",
+                                       "container/non-existentdir/non-existentsubdir"));
+  ASSERT_RAISES(IOError, fs_->CopyFile("container/somedir/subdir/subfile",
+                                       "nonexistent-container/non-existentdir"));
+}
+
+TEST_F(TestAzureFileSystem, OpenInputStream) {
+  std::shared_ptr<io::InputStream> stream;
+  std::shared_ptr<Buffer> buf;
+
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("container"));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("nonexistent-container"));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream(""));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("container/somedir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("container/non-existentdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("container/somedir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("container/non-existentdir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputStream("nonexistent-container/somedir"));
+
+  // "Files"
+  ASSERT_OK_AND_ASSIGN(stream, fs_->OpenInputStream("container/somefile"));
+  ASSERT_OK_AND_ASSIGN(buf, stream->Read(2));
+  AssertBufferEqual(*buf, "so");
+  ASSERT_OK_AND_ASSIGN(buf, stream->Read(5));
+  AssertBufferEqual(*buf, "me da");
+  ASSERT_OK_AND_ASSIGN(buf, stream->Read(5));
+  AssertBufferEqual(*buf, "ta");
+  ASSERT_OK_AND_ASSIGN(buf, stream->Read(5));
+  AssertBufferEqual(*buf, "");
+}
+
+TEST_F(TestAzureFileSystem, OpenInputFile) {
+  std::shared_ptr<io::RandomAccessFile> file;
+  std::shared_ptr<Buffer> buf;
+
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("nonexistent-container"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile(""));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container/somedir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container/non-existentdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container/somedir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("container/non-existentdir/subdir"));
+  ASSERT_RAISES(IOError, fs_->OpenInputFile("nonexistent-container/somedir"));
+
+  // "Files"
+  ASSERT_OK_AND_ASSIGN(file, fs_->OpenInputFile("container/somefile"));
+  ASSERT_OK_AND_EQ(9, file->GetSize());
+  ASSERT_OK_AND_ASSIGN(buf, file->Read(4));
+  AssertBufferEqual(*buf, "some");
+  ASSERT_OK_AND_EQ(9, file->GetSize());
+  ASSERT_OK_AND_EQ(4, file->Tell());
+
+  ASSERT_OK_AND_ASSIGN(buf, file->ReadAt(2, 5));
+  AssertBufferEqual(*buf, "me da");
+  ASSERT_OK_AND_EQ(4, file->Tell());
+  ASSERT_OK_AND_ASSIGN(buf, file->ReadAt(5, 20));
+  AssertBufferEqual(*buf, "data");
+  ASSERT_OK_AND_ASSIGN(buf, file->ReadAt(9, 20));
+  AssertBufferEqual(*buf, "");
+
+  char result[10];
```

Comment:
Could you use `20` instead of `10` because the below `RaedAt()` uses `20` for `nbytes`.